### PR TITLE
[balances/#56] benchmark balances pallet

### PR DESCRIPTION
**Pull request by pifragile**
benchmarks can be tested without building the node by running

```cargo test --features runtime-benchmarks --manifest-path balances/Cargo.toml -- benchmarking```

from the `pallets` repo root.

### CODE REVIEW

File: balances/Cargo.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ authors = ["encointer.org <alain@encointer.org> and Parity Technologies <admin@p
 edition = "2021"
 
 [dependencies]
+approx = "0.5.0"
 codec = { package = "parity-scale-codec", version = "2.3.1", default-features = false, features = ["derive"] }
```
Comment (clangenb) [inline balances/Cargo.toml:9]: As `approx` is only used in the benchmarks, it should be an optional dependency and enable in with the `benchmark` feature.

File: balances/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	balances::{BalanceType, Demurrage},
+	fixed::traits::LossyInto,
+};
+use frame_benchmarking::{account, benchmarks, impl_benchmark_test_suite};
+use frame_support::parameter_types;
+use frame_system::RawOrigin;
+
+parameter_types! {
+	pub const DefaultDemurrage: Demurrage = Demurrage::from_bits(0x0000000000000000000001E3F0A8A973_i128);
+}
```
Comment (clangenb) [inline balances/src/benchmarking.rs:13]: What do you need this for?

**Review by clangenb — COMMENTED** [review]
Looks good in general, only minor comments.

File: balances/src/benchmarking.rs (hunk 2 above)
Comment (pifragile) [reply balances/src/benchmarking.rs]: nothing😂

**Review by clangenb — APPROVED** [review]
Looks good to me

**Review by brenzi — COMMENTED** [review]
you call the benchmark test `transfer` but you are actually benchmarking `issue`. shouldn't that be two distinct benchmarks?

**Comment by clangenb** [comment]
No, issue is still in the setup phase, the benchmarked code is this line:

`_(RawOrigin::Signed(alice.clone()), bob_lookup, cid, BalanceType::from_num(10i32))`

the `_` is repblaced with the benchmark name, and will therefore execute the `transfer` dispatchable.

**Review by brenzi — APPROVED** [review]

**Pull request merged**